api/types/appserver.go

```diff
@@ -295,16 +296,90 @@ func (s *AppServerV3) MatchSearch(values []string) bool {
 	return MatchSearch(nil, values, nil)
 }
 
-// AppServers represents a list of app servers.
-type AppServers []AppServer
+type AppServerSorter struct {
+	servers []AppServer
+	lessFn  func(i, j int) bool
+}
+
+// AppServers returns a sorter that implements the Sort interface,
+// Call its Sort method to sort the data by sort criteria.
+func AppServers(servers []AppServer) *AppServerSorter {
+	return &AppServerSorter{
+		servers: servers,
+	}
+}
+
+// Len is part of sort.Interface.
+func (s *AppServerSorter) Len() int { return len(s.servers) }
+
+// Less is part of sort.Interface.
+func (s *AppServerSorter) Less(i, j int) bool { return s.lessFn(i, j) }
+
+// Swap is part of sort.Interface.
+func (s *AppServerSorter) Swap(i, j int) { s.servers[i], s.servers[j] = s.servers[j], s.servers[i] }
 
-// Len returns the slice length.
-func (s AppServers) Len() int { return len(s) }
+// Sort sorts a list of app servers according to the sort criteria.
+func (s *AppServerSorter) Sort(sortBy *SortBy) error {
```

Comment:
Perhaps `SortBy` and the combination `SetCustomSortFn`/`sort.Sort` can be a `SortByCustom`?